### PR TITLE
Remove requires from individual components

### PR DESCRIPTION
**Pull request by Pontus4**
Solve #85

### CODE REVIEW

File: lib/twingly/amqp/subscription.rb [hunk 1]
```diff
@@ -1,4 +1,3 @@
-require "twingly/amqp/connection"
 require "twingly/amqp/message"
```
Comment (walro) [inline lib/twingly/amqp/subscription.rb:1]: noticed this was kept, is `message` only used here? Or should it be moved to the mega-require too?

Comment (Pontus4) [reply lib/twingly/amqp/subscription.rb:1]: Only used there :)

**Comment by walro** [comment]
I noticed we moved `json`, we have `bunny` lurking around too:

```
$ ag "require \""  lib
...
lib/twingly/amqp/session.rb
1:require "bunny"
...
```

**Comment by Pontus4** [comment]
> I noticed we moved json, we have bunny lurking around too

Same thing there, `bunny` is only used in session, but `json` was required in three different files. 
But it's maybe cleaner to require everything in amqp?

**Comment by walro** [comment]
> But it's maybe cleaner to require everything in amqp?

Good question, I think we should pick one way at least. Might be simpler to move everything into the "main file" as I think that will give a clearer signal on what to do when adding a new dependency.

**Comment by Pontus4** [comment]
> Might be simpler to move everything into the "main file" as I think that will give a clearer signal on what to do when adding a new dependency.

Sure! Both options seem ok to me :)

File: lib/twingly/amqp.rb [hunk 2]
```diff
@@ -8,8 +8,11 @@
 require "twingly/amqp/default_exchange_publisher"
 require "twingly/amqp/topic_exchange_publisher"
 require "twingly/amqp/null_logger"
+require "twingly/amqp/message"
 
+require "bunny"
 require "ostruct"
+require "json"
```
Comment (walro) [inline lib/twingly/amqp.rb:15]: *nitpick*, alphabetical order is nice:)

Comment (Pontus4) [reply lib/twingly/amqp.rb:15]: Should I order the ones above (in a separate commit) as well?

Comment (walro) [reply lib/twingly/amqp.rb:15]: That's fine, this stood out as the "problem" was introduced now. Re-ordering the ones above will ruin the `git blame` in a way, so let's not bother.

Comment (Pontus4) [reply lib/twingly/amqp.rb:14]: Ahh, okay!

**Review by walro — APPROVED** [review]
LGTM!

**Pull request merged**